core/src/main/java/io/grpc/internal/TransportSet2.java

```diff
@@ -80,7 +79,9 @@
   private final Callback callback;
   private final ClientTransportFactory transportFactory;
   private final ScheduledExecutorService scheduledExecutor;
-  private final Executor appExecutor;
+
+  // A serializing executor shared across the Channel
+  private final Executor channelExecutor;
```

Comment:
We have not yet decided the type of the "Channel Executor". It may not necessarily `SerializingExecutor`, because we have not found a satisfactory Executor to base it on.

I considered a SerializingExecutor based on the app executor, but it seems abusive because the app executor is intended for app logic, not for channel bookkeeping. We don't want channel bookkeeping logic to contend for thread resources with app logic, which may increase latency or even starvation.

@ejona86 talked about using a thread-less Executor, which sounds like a better options for me. I'd like to come back to this topic after I finished the refactor.